### PR TITLE
added trailing stop loss rule

### PR DESCRIPTION
**Pull request by Chr1st0ph**
Changes proposed in this pull request:

    add a trailing stop-loss rule ( similar to StopLossRule )
    added an entry to the unreleased section of CHANGES.md

### CODE REVIEW

**Comment by Chr1st0ph** [comment]
I still found a flaw if this rule is combined with other rules. I am working on a fix

**Comment by Chr1st0ph** [comment]
I added another commit to handle combined rules. Please review

File: ta4j-core/src/main/java/org/ta4j/core/Trade.java [hunk 1]
```diff
@@ -102,7 +102,8 @@ public Order getExit() {
     public boolean equals(Object obj) {
         if (obj instanceof Trade) {
             Trade t = (Trade) obj;
-            return entry.equals(t.getEntry()) && exit.equals(t.getExit());
+            return (entry == null ? t.getEntry() == null : entry.equals(t.getEntry()))
+            		&& (exit == null ? t.getExit() == null : exit.equals(t.getExit()));
         }
         return false;
     }
```
Comment (edlins) [inline ta4j-core/src/main/java/org/ta4j/core/Trade.java:109]: Should this throw a `ClassCastException` if not `instanceof Trade`?  Also would be nice to see at least one new test case in the appropriate test class for any change to a pervasive core class.

Comment (team172011) [reply ta4j-core/src/main/java/org/ta4j/core/Trade.java:109]: Test case would be appropriate. If obj is not instance of Trade the equals function returns false, that is standard behaviour of equals function

Comment (Chr1st0ph) [reply ta4j-core/src/main/java/org/ta4j/core/Trade.java:109]: equals-method-policy is defined in java.lang.Object#equals. I added some test-cases...

**Comment by team172011** [comment]
Thanks! Resolved conflicts and merged local with [ef4c2d4](https://github.com/ta4j/ta4j/commit/ef4c2d4bca750e9aecdb6a4c4ee1b91126f2fb69)

**Pull request merged**